File: solutions/HJK9810/2026-08-12/medium/1215.py
def check_palindrome(word):
    words = list(word)
    left, right = 0, len(words) - 1
    while left < right:
        if words[left] != words[right]: return False
        left += 1
        right -= 1
    
    return True

def algorithm():
    N = int(input())
    boards = [list(input()) for _ in range(8)]
    count = 0

    for row in range(8):
        for col in range(8):
            if col <= 8 - N:
                first_word = boards[row][col:col + N]
                if check_palindrome(first_word): count += 1
            if row <= 8 - N:
                second_word = [boards[r][col] for r in range(row, row + N)]
                if check_palindrome(second_word): count += 1

    return count

# T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, 11):
    print(f"#{test_case} {algorithm()}")

File: solutions/HJK9810/2026-08-12/medium/1215.java
import java.util.Scanner;
import java.util.Arrays;

class Solution {
  private static int BOARD_SIZE = 8;

  private boolean checkPalindrome(char[] words) {
    int left = 0;
    int right = words.length - 1;

    while (left < right) {
      if (words[left] != words[right]) return false;
      left++;
      right--;
    }

    return true;
  }

  public static void main (String args[]) throws Exception {
    Solution sol = new Solution();
    Scanner sc = new Scanner(System.in);
    // int T = sc.nextInt();

    for (int test_case = 1; test_case <= 10; test_case++) {
      int N = sc.nextInt();
      char[][] boards = new char[BOARD_SIZE][BOARD_SIZE];
      int count = 0;

      for (int row = 0; row < BOARD_SIZE; row++) {
        char[] alph = sc.next().toCharArray();
        for (int col = 0; col < BOARD_SIZE; col++) {
          boards[row][col] = alph[col];
        }
      }

      for (int row = 0; row < BOARD_SIZE; row++) {
        for (int col = 0; col < BOARD_SIZE; col++) {
          if (col <= BOARD_SIZE - N) {
            char[] rowWord = Arrays.copyOfRange(boards[row], col, col + N);
            if (sol.checkPalindrome(rowWord)) count++;
          }

          if (row <= BOARD_SIZE - N) {
            char[] colWord = new char[N];
            for (int idx = 0; idx < N; idx++) {
              colWord[idx] = boards[row + idx][col];
            }

            if (sol.checkPalindrome(colWord)) count++;
          }
        }
      }

      System.out.printf("#%d %d\n", test_case, count);
		}
	}
}
